### ISSUE TITLE
Publish layer to Gov Cloud regions

### ISSUE THREAD

**Issue by bryantbiggs** [aws-powertools/powertools-lambda-python#1166]
**Is your feature request related to a problem? Please describe.**
- Consider publishing the layer to the Gov Cloud regions. I checked the [list of regions](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/docs/index.md#lambda-layer) the layer is currently published to but it looks like the Gov Cloud regions are not present. 

**Describe the solution you'd like**
- It would be great if the layer could be published and available in the Gov Cloud regions as well

**Describe alternatives you've considered**
- For now, I am recommending Gov Cloud users perform a copy of the layer to their account and use that route as a temporary work around

**Additional context**

**Comment by kb-fusus** [comment]
any updates on this one?

**Comment by rubenfonseca** [comment]
Hi @kb-fusus I'm actively looking for an update on this, and will get back to you shortly. Publishing the layer to the Gov Cloud regions is a very complex operation, but we're looking to understand the best way to making it work. Thank you for your ping!

**Comment by dannellyz** [comment]
@rubenfonseca Also wanted to +1 from a customer side how great it would be to have this available in GovCloud. Specifically, the added optimization the team does compared to just pip installing or packaging ourselves would be a huge win. Thanks!

**Comment by heitorlessa** [comment]
thanks a lot @dannellyz! We're still looking for US specific controls and procedures to work on this.

You could help us prioritize this by sending an email from your corporate address to aws-lambda-powertools-feedback at amazon dot com. We're collecting customer names / gov agencies to help expedite some of these controls.

Thank you!

**Comment by dannellyz** [comment]
Will do!

**Comment by heitorlessa** [comment]
Adding this to our public roadmap as of now to make it easier for agencies to reach out to us to help prioritize: https://docs.powertools.aws.dev/lambda/python/stage/roadmap/#lambda-layer-in-govcloud-and-china-region

Sent an internal email asking about GovCloud region requirements to publish this Layer.

**Comment by leandrodamascena** [comment]
Adding the `internal` label because we need to handle it internally before making it available for the public.

**Comment by leandrodamascena** [comment]
Hi everyone, we're still researching this feature. Because of the complexities of GovCloud, this is taking us longer than expected. We'll share any updates as we have them.

Thank you

**Comment by rubenfonseca** [comment]
Update: we found a way to move forward and to add the Layer to GovCloud regions. Expect to have news by the end of the week! Stay tunned :)

**Comment by rubenfonseca** [comment]
Update: we’ve completed our infrastructure changes and we are going through additional security and compliance requirements. We’ll keep you posted as soon as we have updates on a plausible ETA; tentatively after re:invent / early next year.

**Comment by drissamri** [comment]
It would be really interesting to read on the additional measures you have to take to qualify for the requirements

**Comment by heitorlessa** [comment]
it's all internal @drissamri the only extra bit we will need to do is to create provenance for the Layer w/ all dependencies, or SBOM (easy). 

If you'd like to learn more about AppSec & supply chain security in general, reach out to @sthulb (Simon is on Discord!)

**Comment by leandrodamascena** [comment]
Hey, everyone! We are still working on this with internal teams and we may have some good news next month.

Changing the status to "Working on it"

**Comment by marlhammer** [comment]
greetings! we have a production system that relies on power tools and this is the only blocker to migrating this to GovCloud. :-) can we support this in any way?

**Comment by sthulb** [comment]
Hi folks!

We've been making steps towards our releases on GovCloud (us-gov-west-1, us-gov-east-1). If all goes well, we aim to have releases by the end of July. We'll post weekly updates here to document progress (or lack of it, if we have no new news to share).

Thanks
@sthulb

**Comment by sthulb** [comment]
Hi folks! 

We have created the accounts and in the process of bootstrapping them, hopefully this time next week we'll be in a position to deploy into them.

A question for anyone reading: Do you want:
  - all previous versions backfilled from v2
  - last X versions backfilled
  - just the latest version?

Thanks
@sthulb

**Comment by bryantbiggs** [comment]
just the latest - motivation for folks to update 😉

**Comment by sthulb** [comment]
Hi folks, 

We hit a bit of a block this week with other work taking priority, we'll be back to it today with hopefully something to show for it next week.

Thanks
@sthulb

**Comment by SuzakuTheKnight** [comment]
Really looking forward to the GovCloud layer release. Just the latest version would be enough for our needs.

**Comment by sthulb** [comment]
Hi folks, 

We're back at it! The last two weeks I've been away on PTO and other travel for work. This will be my top priority next week.

Thanks
@sthulb

**Comment by jesseadams** [comment]
This is exciting! How is it going @sthulb ?

**Comment by leandrodamascena** [comment]
Hello everyone! We have already deployed the necessary things we need in GovCloud, like OIDC Role, IAM Roles to deploy layers and others. We will start testing with GH Actions + GovCloud and after we release V3 - which is expected in the next 2 weeks - we hope to publish these layers in GovCloud.

Almost there 🚀

**Comment by leandrodamascena** [comment]
Hello everyone! We released [v3.0.0](https://github.com/aws-powertools/powertools-lambda-python/releases/tag/v3.0.0) two days ago and are now working on creating the GitHub Action to publish Lambda layers to the GovCloud partition. You can expect to see this layer in the next two weeks! Many thanks to everyone who gave us some feedback to help prioritize internally and for your great patience while we unblocked things.

**Comment by sthulb** [comment]
I've created this action: https://github.com/aws-powertools/actions/blob/main/.github/workflows/layer_govcloud.yml which works in beta right now. The plan is to move this to prod soon.

**Comment by maxrabin** [comment]
Hi, piggybacking here to ask about AWS China partition layer. I don't see anything on the 3.x Roadmap for China (though I do see on the 2.x Roadmap).
I'd love to have a published Lambda Layer for AWS China (both Beijing and Ningxia regions).

**Comment by leandrodamascena** [comment]
> Hi, piggybacking here to ask about AWS China partition layer. I don't see anything on the 3.x Roadmap for China (though I do see on the 2.x Roadmap). I'd love to have a published Lambda Layer for AWS China (both Beijing and Ningxia regions).

Hi @maxrabin! This is something we've been looking into and want to add support for. Can you please open a new issue and tell us a bit about your use case? It can be very superficial, it doesn't need to be super detailed or involve anything confidential. This helps us prioritize internally with requests coming from customers.

Thanks

**Comment by maxrabin** [comment]
> > Hi, piggybacking here to ask about AWS China partition layer. I don't see anything on the 3.x Roadmap for China (though I do see on the 2.x Roadmap). I'd love to have a published Lambda Layer for AWS China (both Beijing and Ningxia regions).
> 
> Hi @maxrabin! This is something we've been looking into and want to add support for. Can you please open a new issue and tell us a bit about your use case? It can be very superficial, it doesn't need to be super detailed or involve anything confidential. This helps us prioritize internally with requests coming from customers.
> 
> Thanks

Hi @leandrodamascena I opened #5269

**Comment by switch-to-gitlab** [comment]
@leandrodamascena @sthulb Hello, I see this is pending release. If you don't mind my asking, do you know when we'll see this in our production environments? 

I appreciate your work and don't mean to sound impatient! It's just that I rolled the dice and built this dependency into a user story for this sprint, so I want to know if I can deploy to dev on Monday. 😅

**Comment by sthulb** [comment]
H folks

The issue was closed by virtue of merging a PR, I've reopened it.

We've merged the GovCloud workflow into the develop branch last week and successfully deployed into the GovCloud gamma/beta accounts. The workflow can be observed here: https://github.com/aws-powertools/powertools-lambda-python/actions/workflows/layer_govcloud.yml

Our next steps are to create an internal run book and update the public docs before creating a production release which will hopefully be completed soon. 

Thanks everyone for sticking with this.

**Comment by sthulb** [comment]
We have published GovCloud layers into our PROD account, you can now use them.

You can see the ARN structure here: 
https://docs.powertools.aws.dev/lambda/python/latest/#lambda-layer-govcloud

Thanks everyone, I'll leave this open until we're 100% done, we're in the final stretches of it, but the rest is internal.
